src/compiler/program.ts

```diff
@@ -2229,4 +2226,14 @@ namespace ts {
         Debug.assert(names.every(name => name !== undefined), "A name is undefined.", () => JSON.stringify(names));
         return names;
     }
+
+    function getModuleImportAndAugmentationNames({ imports, moduleAugmentations }: SourceFile): string[] {
+        const res = imports.map(i => i.text);
+        for (const aug of moduleAugmentations) {
+            if (aug.kind === SyntaxKind.StringLiteral) {

```

Comment:
What about the augmentations that are identifiers?

Reply:
We're interested in the names of modules that have to be resolved; `declare global` doesn't require any module resolution.

Reply:
Unless that's obvious to someone more experience, please consider adding a comment to that effect.

Reply:
Done.